backend/app/api/api_v1/routers/documents.py

```diff
@@ -162,3 +173,84 @@ async def post_association(
         document_association.name,
         document_association.type,
     )
+
+
+@documents_router.post(
+    "/document-relationship", response_model=RelationshipEntityResponse, status_code=201
+)
+async def post_relationship(
+    request: Request,
+    relationship: RelationshipCreateRequest,
+    db=Depends(get_db),
+    current_user=Depends(get_current_active_superuser),
+):
+    """Create a relationship"""
+    return create_relationship(
+        db,
+        relationship.name,
+        relationship.type,
+        relationship.description,
+    )
+
+
+@documents_router.get("/document-relationship", response_model=RelationshipGetResponse)
+async def get_relationship(
+    request: Request,
+    db=Depends(get_db),
+    current_user=Depends(get_current_active_superuser),
+):
+    """Get all relationsgips"""
+    return get_relationships(db)
+
+
+@documents_router.get(
+    "/document-relationship/{relationship_id}",
+    response_model=RelationshipAndDocumentsGetResponse,
+)
+async def get_relationship_documents(
+    request: Request,
+    relationship_id: int,
+    db=Depends(get_db),
+    current_user=Depends(get_current_active_superuser),
+):
+    """Get a single relationship and all documents"""
+    return RelationshipAndDocumentsGetResponse(
+        documents=get_documents_in_relationship(db, relationship_id),
+        relationship=get_relationship_by_id(db, relationship_id),
+    )
+
+
+@documents_router.post(
```

Comment:
If we know all the details of the thing we're creating, this should really be a `PUT` not a `POST`

Reply:
Done - I've also ensured its idemopotent